### DISCUSSION TITLE
2808 (Publicando o pacote cm GitHubActions) - Erro ao publicar o pacote

### DISCUSSION THREAD

**Discussion by Leandromenezes81** [balta-io/forum#315] [Tira Dúvidas] [answered]
Ao tentar publicar o pacote, estou capturando o seguinte erro:

`Run dotnet nuget push "src/MoneyExtension/bin/Release/*.nupkg" -k *** -s https://api.nuget.org/v3/index.json
Pushing MoneyExtensions.1.0.0.nupkg to 'https://www.nuget.org/api/v2/package'...
  PUT https://www.nuget.org/api/v2/package/
  Forbidden https://www.nuget.org/api/v2/package/ [6](https://github.com/Leandromenezes81/money-extension/actions/runs/4743727356/jobs/8423650186#step:7:7)24ms
error: Response status code does not indicate success: 403 (The specified API key is invalid, has expired, or does not have permission to access the specified package.).`

Assim está meu arquivo `master.yml`:

`name: Build and deploy package

on:
  push:
    branches:
      - master

jobs:
  build-and-deploy:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@master

      - name: Set up .NET Core
        uses: actions/setup-dotnet@v3
        with:
          dotnet-version: '7.0.x'

      - name: Build with dotnet
        run: dotnet build --configuration Release

      - name: Test
        run: dotnet test

      - name: Create Package
        run: dotnet pack --configuration Release

      - name: Publish
        run: dotnet nuget push "src/MoneyExtension/bin/Release/*.nupkg" -k ${{ secrets.NUGET_TOKEN }} -s https://api.nuget.org/v3/index.json
`

Dúvida: O nome do arquivo yml precisa corresponder ao nome da branch?

Esse é o link do repositório: [https://github.com/Leandromenezes81/money-extension](https://github.com/Leandromenezes81/money-extension)

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @Leandromenezes81, como vai?

O erro abaixo se refere ao seu token de publicação no NuGet que expirou:
`403 (The specified API key is invalid, has expired, or does not have permission to access the specified package.).`

Sobre o nome do arquivo corresponder a branch, não precisa, embora seja uma boa prática. O que manda é este trecho do yml:
```yml
on:
push:
branches:
- master
```

**Reply by Leandromenezes81** [reply]
@andrebaltieri, muito obrigado pelo retorno.

Eu havia criado a API_KEY com prazo de expiração de 365 dias. Mesmo assim, excluí esta e criei outra. no arquivo `master.yml`, mudei o comando de Publish:

`- name: Publish
        run: dotnet nuget push "src/MoneyExtension/bin/Release/*.nupkg" --api-key ${{ secrets.NUGET_TOKEN }} --source https://api.nuget.org/v3/index.json`

Mesmo assim, não funcionou.

**Reply by andrebaltieri** [reply]
A chave que você gerou é um PAT? Foi criado por um usuário que tem acesso ao repositório?

**Reply by Leandromenezes81** [reply]
> A chave que você gerou é um PAT? Foi criado por um usuário que tem acesso ao repositório?

Olha, a primeira pergunta eu preciso pesquisar. Mas não recordo dessa referência PAT. O e-mail da conta do GitHub e da Microsoft são os mesmos.

**Reply by andrebaltieri** [reply]
Bom dia, @Leandromenezes81, como vai?
Nas aulas eu explico como gerar o PAT, por favor, dá uma revisada lá, se houver dúvidas, ai retorna aqui.

**Comment by stRafa** [comment]
Estou com o msm B.O., minha api key está dando erro `error: Response status code does not indicate success: 403 (The specified API key is invalid, has expired, or does not have permission to access the specified package.).`

**Reply by andrebaltieri** [reply]
Bom dia, @stRafa, como vai?

Por favor, dá uma olhada nas outras questões que temos aqui:
https://github.com/balta-io/forum/discussions?discussions_q=403

Neste link tem as dúvidas relacionadas e como resolver seu problema.

**Reply by stRafa** [reply]
Vlw Balta

**Comment by sol-2023** [comment]
![image](https://github.com/balta-io/forum/assets/136741611/7d77bf55-b4f2-4916-80fe-44d1cde73400)
Boa tarde Balta. 
Estou com pb no mesmo módulo, MoneyExtension , só q ele diz q tem erro na linha 7 do workflow.
Eu envio os pacotes mas não publica. Essa é a versão de troca de nome de variavel , mas mesmo na 1a versão, não funcionou. 
Voce sabe o que pode ser ?
Repositorio : https://github.com/sol-2023/money-extensions/actions/runs/5696484306

**Reply by andrebaltieri** [reply]
Bom dia, @sol-2023, consegue checar se a identação do arquivo está OK? Pode ser isto.

**Comment by sol-2023** [comment]
Bom dia Balta,
Abaixo, o arquivo. Não acho q seja identação, mas..
Uma dúvida, eu entro no Nuget com o meu hotmail. Será q é por isso  q não
funciona ?
No módulo seguinte (MyCustomPrivatePackage) também deu erro de publicação.
Percebi que para acessar  o
https://nuget.pkg.github.com/sol-2023/index.json, ele pede a senha da
Microsoft, mas o usuario sol-2023 é do GitHub, que é diferente da conta da
Microsoft (que é o hotmail). Tem que criar uma conta com mesmo
usuario/senha do GitHub ?

Solange
------- o main.yml
name: Build and deploy package
on:
    push:
        branches:
            - main
jobs:
    build-and-deploy:
        runs-on: ubuntu-latest

        steps:
        - uses: ***@***.***

        - name: Set up .NET Core
        uses: ***@***.***
        with:
            dotnet-version: '7.0.x'

        - name: Build with dotnet
        run: dotnet build --configuration Release

        - name: Test
        run: dotnet test

        - name: Create the package
        run: dotnet pack --configuration Release

        - name: Publish
        run: dotnet nuget push "MoneyExtension/bin/Release/*.nupkg" -k ${{
secrets.NUGET_TOKEN }} -s https://api.nuget.org/v3/index.json

<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Não
contém vírus.www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

Em seg., 31 de jul. de 2023 às 10:00, Andre Baltieri <
***@***.***> escreveu:

**Reply by andrebaltieri** [reply]
@sol-2023 acredito que o PAT (Public Access Token) igual eu mostro nas aulas resolve este problema que você menciona de senhas.
Mas se fosse erro de autenticação, você teria uma mensagem diferente, um 401 ou 403...

O erro que você mostrou na imagem diz que tem um erro na sua sintaxe YAML, ou seja, algo no arquivo mesmo.

**Comment by sol-2023** [comment]
Bom dia Balta,
Vou procurar essa aula. Obrigada.
Solange.

<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Não
contém vírus.www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

Em ter., 1 de ago. de 2023 às 09:02, Andre Baltieri <
***@***.***> escreveu:
